provisioning/device/src/interfaces.ts

```diff
@@ -188,3 +188,19 @@ export interface TpmSecurityClient {
   activateIdentityKey(key: Buffer, callback: (err: Error) => void): void;
   getRegistrationId(callback: (err: Error, registrationId?: string) => void): void;
 }
+
+/**
+ * @private
+ */
+export interface SymmetricKeyProvisioningTransport extends PollingTransport {
+  setSasToken(sasToken: string): void;
+}
+
+/**
+ * @private
+ * Public API exposed by the Symmetric Key security client object.  This is only useful if you're writing your own security client.
+ */
+export interface SymmetricKeySecurityClient {
+  getRegistrationId(callback?: Callback<string>): Promise<string> | void;
+  createAuthenticationToken(idScope: string, callback?: Callback<SharedAccessSignature>): Promise<SharedAccessSignature> | void;
```

Comment:
>AuthenticationToken [](start = 8, length = 19)

nit: in some places we use "SharedAccessSignature", in others "SasToken" and in others "AuthenticationToken". we should either go with a generic name if we have multiple scenarios in mind, or with the same specific name.

Reply:
Changed this name to createSharedAccessSignature

---
In reply to: [231359554](https://github.com/Azure/azure-iot-sdk-node/pull/418#discussion_r231359554) [](ancestors = 231359554)